File: src/main/java/com/tngtech/jgiven/resolution/ResolutionHandler.java
package com.tngtech.jgiven.resolution;

import com.intellij.psi.PsiField;

import java.util.Objects;

public class ResolutionHandler {

    private ResolutionProvider resolutionProvider = new ResolutionProvider();

    public boolean resolutionMatches(PsiField field, PsiField fieldToSearch) {
        Resolution fieldResolution = resolutionProvider.getResolutionFrom(field);
        Resolution resolution = resolutionProvider.getResolutionFrom(fieldToSearch);
        if (resolution != fieldResolution) {
            return false;
        }
        if (resolution != Resolution.NAME) {
            return field.getType().getPresentableText().equalsIgnoreCase(fieldToSearch.getType().getPresentableText());
        }

        return Objects.equals(
                field.getName(),
                fieldToSearch.getName()
        );
    }
}

File: src/main/java/com/tngtech/jgiven/resolution/ResolutionHandler.kt
package com.tngtech.jgiven.resolution

import com.intellij.psi.PsiField

class ResolutionHandler(
        private val resolutionProvider: ResolutionProvider = ResolutionProvider()
) {

    fun resolutionMatches(field: PsiField, fieldToSearch: PsiField): Boolean {
        val fieldResolution = resolutionProvider.getResolutionFrom(field)
        val resolution = resolutionProvider.getResolutionFrom(fieldToSearch)
        if (resolution !== fieldResolution) {
            return false
        }
        return if (resolution !== Resolution.NAME) {
            field.type.presentableText.equals(fieldToSearch.type.presentableText, ignoreCase = true)
        } else field.name == fieldToSearch.name
    }
}
